File: prob60/tic_tac_toe_3.cs
// 実行時間 10ms
using static System.Console;

internal class Program
{
    static void Main()
    {
        string[] s = new string[5];
        var sc = System.StringComparison.Ordinal;

        for( int i = 0; i < 5; i++ ) {
            s[i] = ReadLine();
        }
        for( int i = 0; i < 5; i++ ) {
            char[] c = new char[5];

            for( int j = 0; j < 5; j++ ) {
                c[j] = s[j][i];
            }
            if( string.Compare( string.Join( "", c ), "OOOOO", sc ) == 0 ) {
                WriteLine( 'O' );
                return;
            } else if( string.Compare( string.Join( "", c ), "XXXXX", sc ) == 0 ) {
                WriteLine( 'X' );
                return;
            }
        }
        WriteLine( 'D' );
        return;
    }
}

File: prob60/tic_tac_toe_3.py
# 実行時間 10ms
def judge(c):
    b = False

    if "".join(c) == "OOOOO":
        b = True
        print("O")
    elif "".join(c) == "XXXXX":
        b = True
        print("X")
    return b

def main():
    s = []

    for _ in [0] * 5:
        s.append(input())
    for i in range(5):
        t = []

        for j in range(5):
            t.append(s[j][i])
        if judge(t) == True:
            return
    print("D")

main()
